drivers/amazonec2/amazonec2.go

```diff
@@ -250,7 +250,7 @@ func (d *Driver) Create() error {
 		}
 
 		for _, s := range subnets {
-			if s.AvailabilityZone == regionZone {
+			if s.AvailabilityZone == regionZone && s.VpcId == d.VpcId {
```

Comment:
Both availability zone and VPC id can be passed as `Filter.N` params in the call to `DescribeSubnets`.